src/signtool/IBM_Crypto.cpp

```diff
@@ -0,0 +1,702 @@
+/* IBM_PROLOG_BEGIN_TAG                                                   */
+/* This is an automatically generated prolog.                             */
+/*                                                                        */
+/* $Source: src/signtool/IBM_Crypto.cpp $                                 */
+/*                                                                        */
+/* OpenPOWER sb-signing-utils Project                                     */
+/*                                                                        */
+/* Contributors Listed Below - COPYRIGHT 2016                             */
+/* [+] International Business Machines Corp.                              */
+/*                                                                        */
+/*                                                                        */
+/* Licensed under the Apache License, Version 2.0 (the "License");        */
+/* you may not use this file except in compliance with the License.       */
+/* You may obtain a copy of the License at                                */
+/*                                                                        */
+/*     http://www.apache.org/licenses/LICENSE-2.0                         */
+/*                                                                        */
+/* Unless required by applicable law or agreed to in writing, software    */
+/* distributed under the License is distributed on an "AS IS" BASIS,      */
+/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or        */
+/* implied. See the License for the specific language governing           */
+/* permissions and limitations under the License.                         */
+/*                                                                        */
+/* IBM_PROLOG_END_TAG                                                     */
+
+#include <unistd.h>
+#include <stdint.h>
+#include <string.h>
+#include <arpa/inet.h>
+
+#include <algorithm>
+
+#include <openssl/ec.h>
+#include <openssl/err.h>
+#include <openssl/ecdsa.h>
+#include <openssl/obj_mac.h>
+#include <openssl/evp.h>
+#include <openssl/pem.h>
+
+#include "IBM_Utils.h"
+#include "IBM_HexString.h"
+#include "IBM_Exception.h"
+
+#include "IBM_Crypto.h"
+
+
+IBM_Crypto::IBM_Crypto( const std::string& p_mode )
+   : m_mode(p_mode)
+{
+    if ( !((m_mode == IBM_Utils::g_MODE_PRODUCTION) || 
```

Comment:
Why isn't m_mode a specific enum that can only be in these two states?  Take care of it at compile-time instead of run-time.